.github/workflows/python-publish.yml

```diff
@@ -0,0 +1,49 @@
+# These workflows will upload a Python Package using Twine when a release is created
+# For more information see: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#publishing-to-package-registries
+
+name: Upload Python Package
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python
+      uses: actions/setup-python@v3
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install setuptools wheel twine build
+    - name: Build distribution
+      run: |
+        cd python/mlcroissant
+        python -m build
+    - name: Publish
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        verify-metadata: true
+        skip-existing: true
+        packages-dir: python/mlcroissant/dist
+        repository-url: https://upload.pypi.org/legacy/
```

Comment:
The mention of "legacy" caught me off-guard, but that's the actual API. The story goes that until somewhat recently, PyPi never really had a proper upload API -- people just reverse-engineered the endpoint in time. A while back, PyPI actually made a proper API, and since it's compatible with the old mostly undocumented functionality, they called it "legacy", a poor choice (IMO). You can find more information here:
- https://stackoverflow.com/questions/47584147/why-is-twine-1-9-1-still-uploading-to-legacy-pypi
- https://peps.python.org/pep-0694/

Reply:
Interesting! Thanks for the explanation.